### ISSUE TITLE
Remove `IQueryable<TAgg>` support from `IRepository<TAgg>`

### ISSUE THREAD

**Issue by marcwittke** [marcwittke/Backend.Fx#126]
Reasons: 
- It's a smell: The real need is for a query model. [Command/Query Separation](https://en.wikipedia.org/wiki/Command–query_separation) advices to use two separate models for command execution and querying. The `IRpository` interface is designed for commands.
- It's introducing problems
  - no single point of responsibility for rehydrating the persisted aggregate. This prevents us from injecting a (sort of) service like DomainEvents right into the aggregate root.
  - Testing against InMemoryPersistence is a lie. Every possible LINQ query will just work on an array, but has a high risk of failing with a real O/R-Mapping-implementation of IQueryable
  - Implementing `IRepository` without O/R-Mapping in place is impossible (e.g. with Dapper)

Also:
- `GetAll()` is a similarily bad idea

But:
- it will break existing applications
- the `AggregateAuthorization` feature (aka "row based security") is making use of this `IQueryable` property

**Comment by flmarkus** [comment]
I would agree, its cleaner to seperate command and query and not to offer query support in IRepository.
But I guess its not that easy to migrate all the apps to alternatives like IView.
And I currently see not a big need of enforcing this.
